src/ATen/function_wrapper.py

```diff
@@ -10,19 +10,26 @@
 # temporary things we cannot handle
 EXCLUDE_PATTERN = "bernoulli.*|normal.*|exponential.*|random.*|arange.*"
 # what has to be done to add a Operation ...
-# 1. if broadcasting, add (non-virtual) broadcast declaration to Type.h and definition to Type.cpp and
-#    rename underlying function
-TYPE_METHOD_DECLARATION_BROADCAST = CodeTemplate("""\
+# 1. if broadcasting or without the full list of arguments, add a non-virtual
+#    declaration under Type.h
+TYPE_METHOD_DECLARATION_NON_VIRTUAL = CodeTemplate("""\
```

Comment:
Similar to some of the comments I made on @gchanan's PR, I think its worth thinking about all of the different permutations of these things that we are starting to generate. For example, we have boradcast declaration, non-virtual declaration, etc. It's starting to obfuscate the simpler model we started with. cc @zdevito who has smarter thoughts about this, I think

Reply:
Here's the new state of affairs:

1. If the method isn't canonical (i.e doesn't have the full list of arguments), it's non-virtual and defined in `TypeMethods.h`
2. Otherwise, if the method broadcasts, it's not virtual and defined in `Type.cpp`. There's a corresponding virtual method with the `s_` prefix that should be overridden by derived types
3. Otherwise, the method is virtual and should be overridden by derived types

So only "canonical" functions with the full list of arguments handle broadcasting.

The prefixes can be a combination of `s_` and `m_` for "same-size" and "Tensor-only" functions.